r/R/install-arrow.R

```diff
@@ -137,3 +136,68 @@ reload_arrow <- function() {
     message("Please restart R to use the 'arrow' package.")
   }
 }
+
+
+#' Download all optional Arrow dependencies
+#'
+#' @param deps_dir Directory to save files into. Will be created if necessary.
+#' Defaults to the value of `ARROW_THIRDPARTY_DEPENDENCY_DIR`, if that
+#' environment variable is set.
+#'
+#' @return `deps_dir`, invisibly
+#'
+#' This function is used for setting up an offline build. If it's possible to
+#' download at build time, don't use this function. Instead, let `cmake`
+#' download them for you.
+#' If the files already exist in `deps_dir`, they will be re-downloaded and
+#' overwritten. Do not put other files in this directory.
+#' These saved files are only used in the build if `ARROW_DEPENDENCY_SOURCE`
+#' is `BUNDLED` or `AUTO`.
+#' https://arrow.apache.org/docs/developers/cpp/building.html#offline-builds
+#'
+#' ## Steps for an offline install with optional dependencies:
+#'
+#' ### On a computer with internet access:
+#' - Install the `arrow` package
```

Comment:
```suggestion
#' - If you don't already have the `arrow` package installed, get this function by
#' `source("https://raw.githubusercontent.com/apache/arrow/master/r/R/install-arrow.R")`
```

Reply:
Oh, I guess you're also relying on the package installation to deliver the download_dependencies.sh and versions.txt scripts?

Reply:
Yes, unless you can think of a better way! As @jonkeane [pointed out](https://github.com/apache/arrow/pull/11001/#discussion_r698528943), it's possible to download those files from github, but protecting against version mismatch (between what's needed by `tools/cpp/` and what's listed in github's `versions.txt`) could be challenging.

Reply:
Yeah that makes sense. I was hoping to avoid the sound of "to install arrow, first install arrow".

Reply:
Yeah.... fair